src/cloud/deploy/static-content-deployment.md

```diff
@@ -8,7 +8,83 @@ functional_areas:
 
-## Minify content
+## Optimize javascript and css content builds
+
+You can use bundling and minification to build optimized javascript and html content for static content deployment.
+
+### Bundle JavaScript files
+
+JavaScript bundling is an optimization technique you can use to reduce the number of server requests for JavaScript files.
+
+For {{site.data.var.ece}} projects, you can use the Magento [Baler](https://github.com/magento/baler) extension to scan generated JavaScript code and create an optimized JavaScript bundle during the build process. Deploying the optimized bundle to your site can reduce the number of network requests when loading your site and improve page load times.
+
+You must install and configure your project to use Baler.
+
+{:.procedure}
+To install and configure Baler:
+
+1. Use the following Baler extension information to [install the the Baler extension]({{ site.baseurl }}/cloud/howtos/install-components.html#install) in a {{site.var.data.ece}} development branch.
+
+   ```text
+   module name: fastly/magento2
+   repository: https://github.com/magento/m2-baler
+   ```
+
+1. In the same environment, update the `config.php` project configuration file with the following settings to enable the javascript bundling options required to use Baler.
+
+   > `config.php`
+
+   ```php
+       'js' => [
+         'merge_files' => '0',
+         'minify_files' => '0',
+         'enable_js_bundling' => '0',
+         'enable_baler_js_bundling' => '1',
+   ```
+
+   {:.bs-callout-tip}
+   If you do not have a copy of the `config.php` file in your development branch, see [Recommended procedure to manage your settings]({{site.baseurl}}/cloud/live/sens-data-over.html#cloud-config-specific-recomm) to learn how to generate and use this file to manage Magento store configuration for Cloud projects.
+
+1. Update your `.magento.env.yaml` environment configuration file to enable the SCD_USE_BALER option during the build process:
+
+   >  `.magento.env.yaml`
+
+   ```yaml
+   stage:
+      build:
+         SCD_USE_BALER:
+            - "index.php"
+   ```
+
+1. Update the .magento.app.yaml `hooks` configuration with the Node Version Manager (nvm) configuration that Baler requires to generate the Javascript bundle.
+
+   > `.magento.app.yaml`
+
+   ```yaml
+   hooks:
+      build: |
+         unset NPM_CONFIG_PREFIX
+         curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.35.2/install.sh | dash
+         export NVM_DIR="$HOME/.nvm"
+         [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+         nvm install --lts=dubnium
+   ```
```

Comment:
Extra context here is helpful so devs know they need to install NVM & Node _before_ running ECE Tools:
```yaml
hooks:
    build: |
        set -e
        
        unset NPM_CONFIG_PREFIX
        curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.35.2/install.sh | dash
        export NVM_DIR="$HOME/.nvm"
        [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
        nvm install --lts=dubnium
        
        php ./vendor/bin/ece-tools build:generate
        php ./vendor/bin/ece-tools build:transfer

```